### PR TITLE
Feature/remodel register hour

### LINKED ISSUE #6: Page for logging hours
Page for logging hours worked.

Needs to separate between user/manager

Redirected here after login

### LINKED ISSUE #66: Remodel dashboard

### PR DESCRIPTION
**Pull request by davidkristoffersen**
# Changes 💎 
* Log hours using the new hours backend 🕚 
* Improve edit and add hours modals 🏅 
* Delete hours 🗑️ 
* Date picker and weekday selector works 📆 
* Make getCurrentUser register update on object ♻️ 
* Validation to registration and edit form ✔️

Fixes #6 
Fixes #66

### CODE REVIEW

File: ClientApp/src/app/services/user.service.ts [hunk 1]
```diff
@@ -16,19 +16,14 @@ export class UserService {
...
-      } catch (e) {
-        return null;
-      }
+    try {
+      const token = this.sessionService.getToken();
+      const user = await this.userProvider.getUser(token);
+      this.setUser(user);
+      return this._user;
+    } catch (e) {
+      return null;
```
Comment (markhilb) [inline ClientApp/src/app/services/user.service.ts:25]: This should be its own function, maybe "refreshCurrentUser" or something.
That way "getCurrentUser" does not query the database when it does not need to 🔥

File: ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-  Validators,
-  ValidatorFn,
-  AbstractControl,
-} from '@angular/forms';
+import { FormBuilder, Validators } from '@angular/forms';
 
 import { UserService } from '../../../services/user.service';
 import { HoursService } from '../../../services/hours.service';
+import { UserProviderService } from '../../../services/api/user-provider.service';
+
```
[the remark this answers is not in the document]
Comment (NWVi) [reply ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts:7]: Don't see what the problem here is? The new line is there to seperate between validators and services.

[the remark this answers is not in the document]
Comment (markhilb) [reply ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts:7]: But there should not be 😤

File: ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts [hunk 3]
```diff
@@ -18,32 +17,42 @@ import { Hours } from 'src/app/models/hours.model';
...
+  ) {}
+
+  hoursRegisterForm = this.fb.group(
+    {
+      date: [, Validators.required],
+      hours: [, [Validators.required, validateRange(0, 24)]],
+      project: ['', Validators.required],
+    },
+    { validators: validateChange.bind(this) },
+  );
```
Comment (markhilb) [inline ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts:52]: Constructor should still be under variables ...

File: ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts [hunk 4]
```diff
@@ -62,68 +71,79 @@ export class RegisterHoursComponent implements OnInit {
...
-      // Set hours to zero if not found in DB
-      hours = 0;
+    if (this.project) {
+      this.hoursRegisterForm.patchValue({
+        date: this.date.getTime(),
+        hours: this.project.Hours,
+        project: { name: this.project.Project },
+      });
+    } else {
+      // no existing project use default values
```
Comment (markhilb) [inline ClientApp/src/app/components/modals/register-hours/register-hours-modal.component.ts:133]: Capital "n" ❗

**Pull request merged**